### ISSUE TITLE
add icount trigger support

### ISSUE THREAD

**Issue by timsifive** [riscv/riscv-isa-sim#1218]
To ratify the debug spec, we need simulator support for icount triggers. qemu has this, but it doesn't have all the other triggers. It would be best to have icount support here as well.

@YenHaoChen, you've contributed a lot of spike trigger work recently. Are you planning to work on icount triggers as well? (I would have written this to you in an email, but I don't have your email address. Feel free to reach out to me: tim at sifive dot com)

**Comment by scottj97** [comment]
I have an icount implementation I can contribute. I’ll need a few days to
rework it for upstreaming.

**Comment by YenHaoChen** [comment]
Yes, I also have an icount implementation and planned to PR. I was waiting for other PRs before firing the icount PR.
Nevertheless, I am fine if @scottj97 wants to contribute.

**Comment by scottj97** [comment]
In that case @YenHaoChen don't wait for mine. I've realized it's going to take a lot of work to upstream mine. It's work that I need to do anyway, so I'll eventually be ready to upstream, but it could take weeks or months.

**Comment by timsifive** [comment]
For the debug spec, I'd like something merged by the end of February. Sounds that is likely to happen based on all the PR activity. I'm happy to help out if that's necessary as well. For now I'll assume that @YenHaoChen will make an icount PR some time this month, that can then be cleaned up during February. Does that sound doable? (@YenHaoChen, I don't know where you are, but will you be affected by Chinese New Year?)

**Comment by YenHaoChen** [comment]
I fire the icount PR.
https://github.com/riscv-software-src/riscv-isa-sim/pull/1220
